docs/DeveloperGuide.md

```diff
@@ -1882,15 +2069,98 @@ testers are expected to do more *exploratory* testing.
+   * Suggested Fixes:
+     * Have a `convertCsvToJson` and `convertJsonToCsv` method to convert the files accordingly.
+     * Have a `convertCsvToJson` and `convertJsonToCsv` button on the GUI to convert files with ease.
+     
+<br>
 
 
+4. **Better Name Validation**
+  * Current Implementation: Currently, names can only contain alphanumeric characters and spaces. (e.g Names such as `Shaquille O'Neal` or `Rohan s/o Mohan` are considered invalid)
+  * Enhancement: Allow special characters such as `'` or `/` to be allowed in the name field.
+  * Reason: To accommodate to more users, and be more inclusive.
+  * Suggested Fixes: 
+    * Update the validation regex to enable special characters specific to names.
 
-## **Appendix: Instructions for manual testing**
+<br>
+
+5. **Task Alert Functionality**
+   * Current Implementation: Currently, although the deadlines of tasks are tracked and color coded, there is no notification function to alert users of upcoming tasks.
+   * Enhancement: Introduce an alert function for tasks, allowing users to set reminders or receive notifications for specific tasks.
+   * Reason: To enhance task management by providing users with timely alerts and reminders for important tasks or deadlines.
+   * Suggested Fixes: 
+     * Have a `setAlert` command for users to set an alarm for their tasks.
+     * Have a `notification` feature to alert users of their upcoming tasks/deadlines. 
+
+<br>
+
+6. **Max Score Function for Graded Tests**
+   * Current Implementation: Currently, users can input any score for any graded test field, and there is no specified upperbound for the Max Score.
+   * Enhancement: Implement a function to set a maximum score for each component of graded tests, ensuring that scores adhere to predefined limits.
+   * Reason: To standardize grading practices and prevent errors or discrepancies in scored assessments.
+   * Suggested Fixes: 
+     * Have a `setMaxScore` command to specify the maximum score for each component of graded tests, preventing entry of scores that exceed these limits.
+
+<br>
+
+7. **Statistics function for Graded Tests**
+   * Current Implementation: Currently, users can only view the graded test scores for their student, with no other features.
+   * Enhancement: Implement a function to calculate the statistics of the graded test field.
+   * Reason: To allow the Avengers to have a better overview of the student's scores.
+   * Suggested Fixes:
+     * Have a `getMean`, `getMedian`, `getMode` command to return the Avenger's session's mean, median and mode for the graded tests.
+
+<br>
+
+8. **Dynamic Session Number**
+   * Current Implementation: Session numbers are immutable. (e.g a Session 5 will remain as Session 5 even though Sessions 1-4 are deleted)
+   * Enhancement: Make the session number dynamic, updating automatically to reflect the current session number even after deleting previous sessions.
```

Comment:
I think no. The idea behind having static session numbers is that it differs from the other features in a sense that each session is labelled a certain session number, rather than by indexing

Reply:
ok, will remove this